kratos/modified_shape_functions/modified_shape_functions.cpp

```diff
@@ -0,0 +1,359 @@
+//    |  /           |
+//    ' /   __| _` | __|  _ \   __|
+//    . \  |   (   | |   (   |\__ `
+//   _|\_\_|  \__,_|\__|\___/ ____/
+//                   Multi-Physics
+//
+//  License:		 BSD License
+//					 Kratos default license: kratos/license.txt
+//
+//  Main authors:    Ruben Zorrilla
+//
+
+// System includes
+
+// External includes
+
+// Project includes
+#include "modified_shape_functions/modified_shape_functions.h"
+
+namespace Kratos
+{
+
+    /// ModifiedShapeFunctions implementation
+    /// Default constructor
+    ModifiedShapeFunctions::ModifiedShapeFunctions(const GeometryPointerType pInputGeometry, const Vector& rNodalDistances) :
+        mpInputGeometry(pInputGeometry), mrNodalDistances(rNodalDistances) {
+    };
+
+    /// Destructor
+    ModifiedShapeFunctions::~ModifiedShapeFunctions() {};
+
+    /// Turn back information as a string.
+    std::string ModifiedShapeFunctions::Info() const {
+        return "Modified shape functions computation base class.";
+    };
+
+    /// Print information about this object.
+    void ModifiedShapeFunctions::PrintInfo(std::ostream& rOStream) const {
+        rOStream << "Modified shape functions computation base class.";
+    };
+
+    /// Print object's data.
+    void ModifiedShapeFunctions::PrintData(std::ostream& rOStream) const {
+        const GeometryPointerType p_geometry = this->GetInputGeometry();
+        const Vector nodal_distances = this->GetNodalDistances();
+        rOStream << "Modified shape functions computation base class:\n";
+        rOStream << "\tGeometry type: " << (*p_geometry).Info() << "\n";
+        std::stringstream distances_buffer;
+        for (unsigned int i = 0; i < nodal_distances.size(); ++i) {
+            distances_buffer << std::to_string(nodal_distances(i)) << " ";
+        }
+        rOStream << "\tDistance values: " << distances_buffer.str();
+    };
+
+    // Returns the input original geometry.
+    const ModifiedShapeFunctions::GeometryPointerType ModifiedShapeFunctions::GetInputGeometry() const {
+        return mpInputGeometry;
+    };
+
+    // Returns the nodal distances vector.
+    const Vector& ModifiedShapeFunctions::GetNodalDistances() const {
+        return mrNodalDistances;
+    };
+
+    // Computes all the shape function values for the positive side.
+    void ModifiedShapeFunctions::GetPositiveSideShapeFunctionsAndGradientsValues(
+        Matrix &rPositiveSideShapeFunctionsValues,
+        std::vector<Matrix> &rPositiveSideShapeFunctionsGradientsValues,
+        Vector &rPositiveSideWeightsValues,
+        const IntegrationMethodType IntegrationMethod) {
+
```

Comment:
may be a candidate for making it virtual pure

Reply:
Done.